### PR TITLE
refactor vep, fastqc, mosdepth, and contest rules

### PR DESCRIPTION
**Pull request by hassanfa**
### This PR:

Fixed:
- Snakemake rule style for `shell` for some rules. part of #636 

### Review and tests: 
- [ ] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

**Review by ashwini06 — COMMENTED** [review]
👌 As you are standardizing the triple quotes, can you please fix other rules as well :)


https://github.com/Clinical-Genomics/BALSAMIC/blob/5685a605bb55a105f3e4f605f8089cb92cbd49eb/BALSAMIC/snakemake_rules/quality_control/multiqc.rule#L74

https://github.com/Clinical-Genomics/BALSAMIC/blob/develop/BALSAMIC/snakemake_rules/quality_control/picard.rule

https://github.com/Clinical-Genomics/BALSAMIC/blob/develop/BALSAMIC/snakemake_rules/quality_control/picard_wgs.rule

https://github.com/Clinical-Genomics/BALSAMIC/blob/develop/BALSAMIC/snakemake_rules/quality_control/sambamba_depth.rule

https://github.com/Clinical-Genomics/BALSAMIC/blob/develop/BALSAMIC/snakemake_rules/variant_calling/mergetype_normal.rule

https://github.com/Clinical-Genomics/BALSAMIC/blob/develop/BALSAMIC/snakemake_rules/variant_calling/mergetype_tumor.rule

https://github.com/Clinical-Genomics/BALSAMIC/blob/develop/BALSAMIC/snakemake_rules/variant_calling/somatic_tumor_only.rule

https://github.com/Clinical-Genomics/BALSAMIC/blob/develop/BALSAMIC/snakemake_rules/variant_calling/split_bed.rule

**Comment by hassanfa** [comment]
@ashwini06 of course!

I created an issue to start refactoring all rules: https://github.com/Clinical-Genomics/BALSAMIC/issues/636 

It is gonna be a tedious process. Any help is welcome :-p

**Comment by ashwini06** [comment]
> @ashwini06 of course!
> 
> I created an issue to start refactoring all rules: #636
> 
> It is gonna be a tedious process. Any help is welcome :-p

Agreed,  I will help you 😃  
Then let's merge this PR (after fixing the changelog), and the rest of the tasks will take in multiple PRs.

**Comment by hassanfa** [comment]
Review comments:

1. benchmark file names
2. order of snakemake fields according to etiquette
3. threads: use correct values

File: BALSAMIC/snakemake_rules/quality_control/fastqc.rule [hunk 1]
```diff
@@ -9,17 +9,24 @@ rule fastqc:
...
-    "mkdir -p ${{tmpdir}}; "
-    "fastqc {input.read1} --dir ${{tmpdir}} --outdir {params.fastqc_dir};" 
-    "fastqc {input.read2} --dir ${{tmpdir}} --outdir {params.fastqc_dir};"
+    """
+source activate {params.conda};
+
+mkdir -p {params.tmpdir};
+export TMPDIR={params.tmpdir};
+
+fastqc {input.read1} --dir {params.tmpdir} --outdir {params.fastqc_dir};
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/quality_control/fastqc.rule:30]: Command splitting to new lines?

File: BALSAMIC/snakemake_rules/quality_control/fastqc.rule [hunk 2]
```diff
@@ -9,17 +9,24 @@ rule fastqc:
...
+fastqc {input.read2} --dir {params.tmpdir} --outdir {params.fastqc_dir};
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/quality_control/fastqc.rule:31]: Command splitting to new lines?

**Review by ashwini06 — APPROVED** [review]
💯

**Pull request merged**